lib/plivo/xml/multipartycall.rb

```diff
@@ -0,0 +1,175 @@
+        elsif !attributes[:coachMode]
+          attributes[:coachMode] = true
+        end
+
+        if attributes[:mute] && !VALID_MUTE_VALUES.include?(attributes[:mute])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:mute]} for mute"
+        elsif !attributes[:mute]
+          attributes[:mute] = false
+        end
+
+        if attributes[:hold] && !VALID_HOLD_VALUES.include?(attributes[:hold])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:hold]} for hold"
+        elsif !attributes[:hold]
+          attributes[:hold] = false
+        end
+
+        if attributes[:startMpcOnEnter] && !VALID_START_MPC_ON_ENTER_VALUES.include?(attributes[:startMpcOnEnter])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:startMpcOnEnter]} for startMpcOnEnter"
+        elsif !attributes[:startMpcOnEnter]
+          attributes[:startMpcOnEnter] = true
+        end
+
+        if attributes[:endMpcOnExit] && !VALID_END_MPC_ON_EXIT_VALUES.include?(attributes[:endMpcOnExit])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:endMpcOnExit]} for endMpcOnExit"
+        elsif !attributes[:endMpcOnExit]
+          attributes[:endMpcOnExit] = false
+        end
+
+        if !attributes[:enterSound]
+          attributes[:enterSound] = 'beep:1'
+        end
+
+        if attributes[:enterSoundMethod] && !VALID_ENTER_SOUND_METHOD_VALUES.include?(attributes[:enterSoundMethod])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:enterSoundMethod]} for enterSoundMethod"
+        elsif !attributes[:enterSoundMethod]
+          attributes[:enterSoundMethod] = 'GET'
+        end
+
+        if !attributes[:exitSound]
+          attributes[:exitSound] = 'beep:2'
+        end
+
+        if attributes[:exitSoundMethod] && !VALID_EXIT_SOUND_METHOD_VALUES.include?(attributes[:exitSoundMethod])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:exitSoundMethod]} for exitSoundMethod"
+        elsif !attributes[:exitSoundMethod]
+          attributes[:exitSoundMethod] = 'GET'
+        end
+
+        if attributes[:onExitActionMethod] && !VALID_ON_EXIT_ACTION_METHOD_VALUES.include?(attributes[:onExitActionMethod])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:onExitActionMethod]} for onExitActionMethod"
+        elsif !attributes[:onExitActionMethod]
+          attributes[:onExitActionMethod] = 'POST'
+        end
+
+        if attributes[:relayDTMFInputs] && !VALID_RELAY_DTMF_INPUTS_VALUES.include?(attributes[:relayDTMFInputs])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:relayDTMFInputs]} for relayDTMFInputs"
+        elsif !attributes[:relayDTMFInputs]
+          attributes[:relayDTMFInputs] = false
+        end
+
+        raise PlivoXMLError, 'No MPC name set for the MPC' unless body
+        super(body, attributes)
+      end
+
```

Comment:
here on add validations for URL fields also
please add new validation helper function if not present in SDK
This function should validate if given is valid URL or not
regex pattern can be taken from python-sdk for reference